### PR TITLE
Remove pod annotation when exiting

### PR DESCRIPTION
**Pull request by mauriciovasquezbernal**
Clean up the "traceloop.kinvolk.io/state" pod annotations that could
have been created before.

Needed for https://github.com/kinvolk/inspektor-gadget/pull/338

### CODE REVIEW

File: traceloop.go [hunk 1]
```diff
@@ -284,7 +284,7 @@ func main() {
 		if err != nil {
 			panic(err)
 		}
-		server.Serve(unixListener)
+		go server.Serve(unixListener)
```
Comment (alban) [inline traceloop.go:287]: I wonder if we should add SIGTERM in addition to SIGINT on line 302. But I see that https://github.com/kinvolk/inspektor-gadget/pull/338 uses SIGINT so this should work.

**Review by alban — APPROVED** [review]
LGTM

File: pkg/straceback/straceback.go [hunk 2]
```diff
@@ -1008,4 +1008,5 @@ func (sb *StraceBack) Stop() {
 		}
 	}
 	sb.mainProg.Close()
+	sb.annotationPublisher.Clean()
```
Comment (alban) [inline pkg/straceback/straceback.go:1011]: Shouldn't you test if `annotationPublisher == nil`  or use `withAnnotationPublisher`?

**Review by alban — CHANGES_REQUESTED** [review]

File: pkg/straceback/straceback.go (hunk 2 above)
Comment (mauriciovasquezbernal) [reply pkg/straceback/straceback.go]: You're right. Thanks for catching it up.

File: traceloop.go (hunk 1 above)
Comment (mauriciovasquezbernal) [reply traceloop.go:289]: I added `SIGTERM` to be on the safe side.

**Pull request merged**